File: examples/javascript/api_call.js
// Calling a remote API in JavaScript

async function fetchData(url) {
    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error("Error:", error.message);
        return null;
    }
}

// Example: Fetch a user from JSONPlaceholder API
const url = "https://jsonplaceholder.typicode.com/users/1";

fetchData(url).then(user => {
    if (user) {
        console.log(`Name: ${user.name}`);
        console.log(`Email: ${user.email}`);
    }
});

File: examples/python/api_call.py
# Calling a remote API in Python
import urllib.request
import json

def fetch_data(url):
    try:
        with urllib.request.urlopen(url) as response:
            data = json.loads(response.read().decode())
            return data
    except Exception as e:
        print(f"Error: {e}")
        return None

# Example: Fetch a user from JSONPlaceholder API
url = "https://jsonplaceholder.typicode.com/users/1"
user = fetch_data(url)

if user:
    print(f"Name: {user['name']}")
    print(f"Email: {user['email']}")
